### ISSUE TITLE
Cut also externally added tracks

### ISSUE THREAD

**Issue by mauticom** [mifi/lossless-cut#896]
**VERSION:** 3.39.0 macOS

**BUG:** When you add an additional audio track to an mp4 file and cut the video and export the new audio track passage is something totally different.

**REPRODUCE:**
1. Open mp4 file
2. Add audio track (longest or shortest, doesn't matter)
3. Cut and extract a part of the video
4. Added audio parts don't fit the time codec

**ADDITIONAL INFO/WORKAROUND:**
If export the complete video with the newly added audio track first everything is fine. So it seems the cut points aren't transfered to the newly added audio file.

Thanks a lot for the great work.

**Comment by mifi** [comment]
I tried `-ss` and `-t` for each file, but it doesn't seem to work 🤔 need to research more.

**Comment by mauticom** [comment]
Thanks for the effort.

**Comment by andoruB** [comment]
I remember distinctly that I was able to do this in older versions of LosslessCut, but now it's no longer possible. Hopefully this will have a resolution.  
For now I'd recommend to mux the files with mkvtoolnix instead.

**Comment by mifi** [comment]
That’s interesting. Do you know which version it used to work? Maybe it worked unintentionally, but I broke it in a newer version when i implemented something else

**Comment by andoruB** [comment]
I'm not sure what version it was, as I long deleted that appimage I used before (wish I hadn't though...)
I think it was dated summer 2020 or thereabouts.

**Comment by cello-tim** [comment]
I've also had problems reliably splicing video/audio clips after adding audio to video clips . But after reading this from the LAME MP3 project I'm thinking it's better to mix and recode the MP3 stream before adding it to the video:

https://lame.sourceforge.io/tech-FAQ.txt (in particular section 4. Why can't MP3 files be seamlessly spliced together?)

TL;DR is that MP3 files are encoded in 576 sample "granules" that contain data that isn't localized to a particular sample (rather a region of samples). It may be possible to splice specially encoded MP3s, but it's not necessarily easy. 

Perhaps it's possible to contemplate having FFMPEG use copy for video streams and select an audio codec for audio streams - but that seems to require more UI to allow for codec options, etc.  Right now I'm happy doing the audio splicing elsewhere...

**Comment by Veryapeman** [comment]
I think i didn't get it. 
I did like you said and it didnt work.

I just want to upload video and put an external audio track on it. Simple, but doesn't work.
Can someone help me out? 

Here is how its happening for me. I tried with another video, but it's the same.
https://www.youtube.com/watch?v=TlbSexG2j9k

Thank you!

**Comment by Veryapeman** [comment]
> I think what's happening here is that when you add an extra track and you also set cut points, it will only cut the main file, so the extra audio track will be included in its full length and will not be cut. I will try to add `-ss` in front of every file added, e.g.: `-ss xx:yy -i file1 -ss xx:yy -i file2`.

I think i didn't get it.
I did like you said and it didnt work.

I just want to upload video and put an external audio track on it. Simple, but doesn't work.
Can someone help me out?

Here is how its happening for me. I tried with another video, but it's the same.
https://www.youtube.com/watch?v=TlbSexG2j9k

Thank you!

**Comment by mifi** [comment]
@Veryapeman you tried to add an mp3 audio to an mp4 file, but mp4 doesn't support mp3 audio.
Try to change output format to mkv, that should work

**Comment by mifi** [comment]
@Veryapeman  I'm not sure exactly what is your issue? The audio you added is not audible in the exported file? If so, you may need to disable the existing audio track, or your file will have two audio tracks

**Comment by wangdh226** [comment]
Hi, just wondering if there is any progress on this ticket? Seems like it has been stale for 1.5 years now...

I just wanted to add my own work around to see if any ideas can come out of it:
Situation - I have recorded some gameplay videos using AMD ReLive (don't think the recording software matters), but I have the separated each clip into 2 different files: 
1. mp4 with game video + game audio
2. m4a with mic audio 

This is because I don't always care about the mic audio, so I want to keep it as a separate track instead of combining it with the game audio. This way I have it just in case I do want the mic audio. However, if I decide I want to keep the audio, I can add the mp4 file and the m4a and trim them with Lossless Cut. 
Herein lies the problem in this ticket (if I'm understanding correctly: When I have both files in Lossless Cut and trim, the mp4 is cut properly, and the mp4 is cut properly based on the timeline, however the m4a file basically just cuts from the first frame with the same length as the mp4. 
This is an issue because the original clips are usually 3 mins long, but sometimes I only want the last 20secs. So cutting them together would correctly give me the last 20secs of the mp4 game video and game audio, but the only the first 20secs of the m4a.

My workaround:
I am using an old version of Lossless Cut because manually entering the frame/time was removed...
  1. Trim mp4 segments to desired time stamps and export
  2. Copy the start and end time stamp elsewhere (notepad, discord chat, browser url, wherever) 
  4. Open m4a audio file, not combine
  5. Copy start timestamp and end timestamp, export m4a audio
  6. Open new trimmed mp4 and combine with new trimmed m4a audio
  7. Export, both files should be ~same length, so its just a merge into a single file with 1 video track and 2 audio tracks

Could this issue be resolved by adding a couple steps to the internal logic/flow when there are extra tracks that also need trimming? Sort of like my workaround?

**Comment by mifi** [comment]
> I am using an old version of Lossless Cut because manually entering the frame/time was removed...

not sure what you mean by that. aren't you able to enter time values into the input boxes near the play button?(see screenshot)

<img width="330" alt="Screenshot 2024-05-03 at 21 29 31" src="https://github.com/mifi/lossless-cut/assets/402547/164abcd5-1fdb-4aa4-a41c-f7ab9ada70d0">

The reason why this has not been prioritized is because it's not trivial to implement _and_ because I believe the same can be achieved with a few extra steps:
1. open file 1 (e.g. video)
2. open file 2 (e.g. audio) and when prompted, choose add tracks from the file
3. export
4. open exported file (which now has video+audio)
5. create segments
6. export again and now both audio+video have been cut

or maybe i'm missing something

**Comment by mauticom** [comment]
Yes, indeed you are missing the point that you still don't want to have the mic audio merged with the game video + audio.

My current workaround is:

1. cut the game video + audio.
2. export the segments
3. export project LLC
4. open mic audio
5. import project LLC
6. export segments (which are no in sync with the game video + audio segements)

From what I understand is that you want(ed) to do this with one FFMPEG command in the background. Wouldn't it be quite easy to add a feature to just apply the segments from the first opened file to all additional added tracks and run the same FFMPEG command that you would use on the file without any additional tracks? This would save me and @wangdh226 and probably many other users lots of time.

Thank you!

**Comment by mifi** [comment]
I think I got it to work now. y'all can test it in the next nightly build.
